src/components/td/td-styles.jsx

```diff
@@ -20,12 +20,61 @@ export default theme => {
       minWidth: 20,
       paddingTop: 8,
       paddingBottom: 8,
+      display: 'block',
+
+      '&::before': {
+        display: 'block',
+        content: 'attr(data-title)',
+      },
 
       [mixins.media('md')]: {
         fontSize: 14,
+        display: 'inline-block',
         minWidth: 40,
         paddingTop: 10,
         paddingBottom: 10,
+        '&::before': {
+          content: 'none',
+        },
+      },
+
+      '&.flexOrder': {
+        order: props => props.flexOrder,
+
+        [mixins.media('md')]: {
+          order: () => 1,
+        },
+      },
+
+      '&.flexBasis': {
+        flexBasis: props =>
+          props.flexBasisMobile
+            ? `${props.flexBasisMobile}${typeof props.flexBasisMobile === 'number' ? '%' : ''}`
+            : `${props.flexBasisDesktop}${typeof props.flexBasisDesktop === 'number' ? '%' : ''}`,
+
+        [mixins.media('md')]: {
+          flexBasis: props => `${props.flexBasisDesktop}${typeof props.flexBasisDesktop === 'number' ? '%' : ''}`,
+        },
+      },
+
+      '&.width': {
+        width: props => `${props.width}${typeof props.width === 'number' ? '%' : ''}`,
+      },
+
+      '&.align': {
+        textAlign: props => (props.alignMobile ? props.alignMobile : props.alignDesktop || props.align),
+
+        [mixins.media('md')]: {
+          textAlign: props => props.alignDesktop || props.align,
+        },
+      },
+
+      '&.hidden': {
+        display: props => (props.hiddenOnMobile ? 'none' : 'block'),
```

Comment:
Another dev mentioned this can be deconstructed:
```
 display: ({hiddenOnMobile}) => (hiddenOnMobile ? 'none' : 'block'),
```

as above, `block` should be `table-cell`, though I'd much prefer finding a way to only set the `display` prop if `hiddenOnMobile` is set, though I can't find examples of such a case.